src/components/Calculator.js

```diff
@@ -1,37 +1,51 @@
-        <div className="result-div">0</div>
-        <Button btnClass="btn-calc" btnValue="AC" />
-        <Button btnClass="btn-calc" btnValue="+/-" />
-        <Button btnClass="btn-calc" btnValue="%" />
-        <Button btnClass="btn-calc btn-orange" btnValue="÷" />
-        <Button btnClass="btn-calc" btnValue="7" />
-        <Button btnClass="btn-calc" btnValue="8" />
-        <Button btnClass="btn-calc" btnValue="9" />
-        <Button btnClass="btn-calc btn-orange" btnValue="x" />
-        <Button btnClass="btn-calc" btnValue="4" />
-        <Button btnClass="btn-calc" btnValue="5" />
-        <Button btnClass="btn-calc" btnValue="6" />
-        <Button btnClass="btn-calc btn-orange" btnValue="-" />
-        <Button btnClass="btn-calc" btnValue="1" />
-        <Button btnClass="btn-calc" btnValue="2" />
-        <Button btnClass="btn-calc" btnValue="3" />
-        <Button btnClass="btn-calc btn-orange" btnValue="+" />
-        <Button btnClass="btn-calc btn-zero" btnValue="0" />
-        <Button btnClass="btn-calc" btnValue="." />
-        <Button btnClass="btn-calc btn-orange" btnValue="=" />
+        <div className="result-div">{output}</div>
+        <Button btnClass="btn-calc" btnValue="AC" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc" btnValue="+/-" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc" btnValue="%" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc btn-orange" btnValue="÷" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc" btnValue="7" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc" btnValue="8" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc" btnValue="9" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc btn-orange" btnValue="x" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc" btnValue="4" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc" btnValue="5" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc" btnValue="6" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc btn-orange" btnValue="-" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc" btnValue="1" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc" btnValue="2" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc" btnValue="3" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc btn-orange" btnValue="+" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc btn-zero" btnValue="0" btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc" btnValue="." btnClickHandler={this.calculateOutput} />
+        <Button btnClass="btn-calc btn-orange" btnValue="=" btnClickHandler={this.calculateOutput} />
```

Comment:
- [ ] It will be better if you use the map function and make an array of objects with btnValues and btnClass and iterate over them instead of repeating code to make your code DRY and follow Javascript best practices.
```js
// complete this array for all buttons, this will be at the top before return.
 const btns = [ {class:"btn-calc",value:'AC'}, {class:"btn-calc",value:'+/-''}, {class:"btn-calc btn-orange",value:'x'}]; 
// This will be after the div instead of all buttons.
{btns.map(({class,value}) => (
            <Button
              btnClickHandler={this.calculateOutput}
              btnClass={class}
              btnValue={value}
              key={value}
            />
          ))}
 
```

Reply:
@omarsalem7,
Changes are done as suggested.